### ISSUE TITLE
Runtime injector does not allow lazy binding to std::function factory methods

### ISSUE THREAD

**Issue by tripleslash** [boost-ext/di#529]
## Expected Behavior

Runtime injector does not allow binding to lambda callbacks

## Steps to Reproduce the Problem

If you wrap the following code in a runtime_injector you will get a compiler error:

```
        auto ctInjector= boost::di::make_injector(
            boost::di::bind<IInterface>()
                    .to([](const auto& injector)
                    {
                        return injector.template create<Implementation>();
                    })
        );
        runtimeInjector.install(ctInjector);
```

## Specifications

  - Version:
  - Platform:
  - Subsystem:

**Comment by BpsTheCool** [comment]
Lots of things that work with normal injector does not work with runtime injector. I've got this problem too with it.

**Comment by kris-jusiak** [comment]
That's true, runtime injector was just an experiment as an extension. but it seems to be popular, hence, let me improve the quality of it, sorry for the delay.

**Comment by BpsTheCool** [comment]
Is there any plan to work on runtime injector? There haven't been any update.
